src/main.rs

```diff
@@ -830,6 +865,117 @@ async fn run() -> Result<(), S2CliError> {
                 writer.flush().await.expect("writer flush");
             }
         }
-    }
+
+        Commands::Ping {
+            basin,
+            stream,
+            interval,
+            batch_bytes,
+            num_batches,
+        } => {
+            let cfg = config::load_config(&config_path)?;
+            let client_config = client_config(cfg.auth_token)?;
+            let stream_client = StreamService::new(StreamClient::new(client_config, basin, stream));
+
+            let interval = interval.max(Duration::from_millis(100));
+            let batch_bytes = batch_bytes.min(50 * 1024);
+
+            eprintln!("Preparing...");
+
+            let mut pinger = Pinger::init(&stream_client).await?;
+
+            let mut pings = Vec::new();
+
+            async fn ping_next(
+                pinger: &mut Pinger,
+                pings: &mut Vec<PingResult>,
+                interval: Duration,
+                batch_bytes: u64,
+            ) -> Result<(), S2CliError> {
+                let jitter_op = if rand::random() {
+                    u64::saturating_add
+                } else {
+                    u64::saturating_sub
+                };
+
+                let record_bytes = jitter_op(batch_bytes, rand::thread_rng().gen_range(0..10));
```

Comment:
the jitter seems too small, you can do like +/- 50% of the target size

Reply:
So 20 KiB +/- 10 KiB ? Isn't that too big?

Reply:
Hmm maybe 25%... want to exercise a wider range of variation. You can also grow the default to 32KiB and max to 128KiB.